Added daemon set

DipeshMitthalal opened the pull request:
Added feature for helm chart to include daemonset
Bumped version of helm chart
This will aid in issue #309

DipeshMitthalal commented:
@richarddli  Can you pleas take a look

DipeshMitthalal commented on helm/ambassador/Chart.yaml:
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 name: ambassador
-version: 0.40.0
+version: 0.40.1
 sources:
```
@richarddli 
Thank you for looking into it!
Ok:-) Should i revert the version or you will?
May i know why it will not be part of 0.40.1?

richarddli commented:
Thanks for working on this! On a quick glance, this looks good. The main comment I have is that we ship standard Kubernetes YAML as well (not just the Helm chart):

https://github.com/datawire/ambassador/tree/master/templates/ambassador

(In fact, most of our users just use the Kubernetes YAML and not the Helm chart). Our regression tests as well use the Kubernetes YAML in general, too. Would you be able to create the Kubernetes YAML equivalent?  This way we can do automated testing of the DaemonSet as well if necessary.

Finally, I'd love some comments (in here is fine) or documentation around your deployment model with a DaemonSet so that we could share it with other people. If you want to add to the documentation directly, you could create a section here: https://github.com/datawire/ambassador/blob/master/docs/reference/running.md.

DipeshMitthalal commented:
@richarddli 
I have updated documentation for daemonset scenario. 
Reverted version bump.
Kubernetes YAML file for daemonset will be lot of duplication with already exisiting YAML files. It's just two lines change which i added to the documentation.

Let me know your thoughts!

nbkrause commented:
Works perfectly with both helm and yaml deploys.

DipeshMitthalal commented on docs/reference/running.md:
```diff
@@ -44,6 +44,14 @@ spec:
+
+* Ideal scenario could be when you are running containers on Kubernetes along side with your non containerized applications running exposed via VIP using BIG-IP or similar products. In such cases, east-west traffic is routed based on iRules to certain set of application pools consisting of application or web servers. In this setup, along side of traditonal application servers, two or more Ambassa ...
+
+* In manifest files `kind: Deployment` nees to be updated to `kind: DaemonSet`  and  `replicas` should be removed in `spec`section.
```
@richarddli Fixed. Thanks for spotting it!

richarddli commented:
Thanks so much!
